### ISSUE TITLE
textDocument/definition not working

### ISSUE THREAD

**Issue by cinghiopinghio** [pappasam/jedi-language-server#52]
Hi nice work,

I have some problems using the goto definition feature.
My setup include `nvim` + `LanguageClient` (or even builtin LSP) and `jedi-language-server`.

The problem in my case is that when calling `LanguageClient#textDocument_definition()` the cursor do not move, `nvim` complains as:
```
[LC] data did not match any variant of untagged enum GotoDefinitionResponse
```

and the client log is as follows:

```17:04:44 DEBUG writer-Some("python") src/rpcclient.rs:254 => Some("python") {"jsonrpc":"2.0","method":"textDocument/definition","params":{"bufnr":1,"character":0,"filename":"/home/mauro/t.py","gotoCmd":null,"handle":true,"languageId":"python","line":4,"method":"textDocument/definition","position":{"character":0,"line":4},"text":["def foo():","    print('FOO')","","","foo()",""],"textDocument":{"uri":"file:///home/mauro/t.py"}},"id":1}
17:04:44 DEBUG reader-Some("python") src/rpcclient.rs:207 <= Some("python") {"id": 1, "jsonrpc": "2.0", "result": [{"uri": null, "range": {"start": {"line": 0, "character": 4}, "end": {"line": 0, "character": 7}}}]}
17:04:44 ERROR unnamed src/rpchandler.rs:27 Error handling message: data did not match any variant of untagged enum GotoDefinitionResponse
Message: {"jsonrpc":"2.0","method":"textDocument/definition","params":{"bufnr":1,"character":0,"filename":"/home/mauro/t.py","gotoCmd":null,"handle":true,"languageId":"python","line":4,"text":["def foo():","    print('FOO')","","","foo()",""]},"id":1}
Error: data did not match any variant of untagged enum GotoDefinitionResponse
```

In particular note that it seams that the server do not send back the URI to the file, instead there is a `null` entry, and that with the method `textDocument/publishDiagnostics`, the server send back the right URI to the file position.

I found the same problem with anakin https://github.com/muffinmad/anakin-language-server/issues/17 but not with other servers such as `python-language-server`.

**Comment by pappasam** [comment]
Thanks for opening this issue and hopefully we can get it resolved soon! It would be helpful if you could provide the following:

1. Your Python version
2. Your Operating System
3. Your jedi-language-server version
4. Your Neovim version
5. The example codebase that you're experiencing this problem with
6. If possible, a minimal vimrc that I can use to reproduce the issue on my machine

At present, I'm not able to immediately reproduce this issue with coc.nvim on my OS (Linux Mint 20) with Python 3.8.6, so I'm guessing the issue might have to do with some deeper difference in our environments.

**Comment by cinghiopinghio** [comment]
It appears that this issue is linked to the development version of `jedi`:

@davidhalter says:

> I know this is kind of breaking the API, but that's just because I'm dropping Python 2 in 0.18.0 and wanted to clean up some API details.


In fact in this version paths are returned as `PosixPath` instead of simple strings.
Taking care of this solved the mentioned issue with `anakinls`.

Python 3.9.1
Archlinux
jedi development version (git)
neovim development version

**Comment by pappasam** [comment]
Ah, you're on the development version of Jedi! I believe the latest version is making some significant changes and I'll be targeting it once these changes have stabilized / been documented a bit better.

Note: the latest version of `jedi-language-server` pins `jedi` to its latest stable version [0.17.2](https://github.com/pappasam/jedi-language-server/blob/master/pyproject.toml#L48). I'll keep this issue open as a reminder that the next version of Jedi has some breaking changes.

In the meantime, I plan to open an issue with `pygls` requesting that `from_fs_path` also accept `pathlib.Path` objects. That would probably be more elegant than adhoc conversion to `str` across this codebase.

**Comment by pappasam** [comment]
On second thought, since this tool targets the latest version of Jedi, I'll just wait for the next version of Jedi and use the path objects directly. `from_fs_path` is unnecessary when working with `pathlib.Path` objects; it provides its own [as_uri](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_uri) method

**Comment by cinghiopinghio** [comment]
Happy to help.

**Comment by pappasam** [comment]
resolved in the latest published release (0.22.0)!
